doc/nrf-bm/migration/nrf5_bm_migration.rst

```diff
@@ -0,0 +1,222 @@
+
+    .. code-block:: cmake
+
+       target_sources(app PRIVATE src/main.c src/other_module.c)
+
+* Include headers:
+
+  * Add directories for custom include paths to ensure header files are correctly found.
+
+    .. code-block:: cmake
+
+       target_include_directories(app PRIVATE include)
+
+* Link libraries:
+
+  * Add external or custom libraries required for your project (such as math, custom vendor libraries).
+
+    .. code-block:: cmake
+
+       target_link_libraries(app PRIVATE libfoo)
+
+* External modules and definitions:
+
+  * If your project relies on external modules or additional functionality, include them.
+
+    .. code-block:: cmake
+
+       add_subdirectory("path_to_directory")
+
+Dependencies are added to the project configuration and incorporated into the build process automatically (you do not have to manually add files from specific locations), with the aid of a tool called Kconfig, in conjunction with CMake.
+That is done using the :file:`prj.conf` file that replaces the :file:`sdk_config.h` file from the nRF5 SDK.
+The entries in the :file:`prj.conf` file are referred to as **Kconfig options**.
+
+Unlike the :file:`sdk_config.h` file that lists all the configuration options relevant for an application, even those whose values are unchanged from defaults, the :file:`prj.conf` file only contains entries whose values are manually set or overriden.
+
+The configuration options whose values are left as default are not present in that file, although when a project is built, a file containing all configuration options pertaining to the application (called :file:`autoconf.h`) is created in the background.
+The build system and the |nRFVSC| extension both provide a way to conveniently browse and search all available project options and inspect their dependencies and read their help text (menuconfig/extension).
+
+There is no distinction or taxonomy between Kconfig options that are applicable to regular |NCS| only and those that are only applicable to |BMlong|.
+Kconfig options that are applicable to the current application are shown and are selectable, while others are not.
+
+There is no consistent mapping of the :file:`sdk_config.h` entries to Kconfig options.
+Some libraries that were ported from nRF5 have similar Kconfig options as the :file:`sdk_config.h` entries they had in nRF5, but it is not a consistent rule.
+
+In nRF5, the same application/sample had a project file for each supported board and IDE/compiler.
+In |BMshort|, there is a single project file (consisting of :file:`CMakeLists.txt` and :file:`prj.conf`) that can be built for different boards with a different command-line instruction or by selecting a different board target in the VS Code extension.
+If necessary, Kconfig options can be specified in a different :file:`.conf` file that is then automatically appended to the default :file:`prj.conf` file, thus realizing a dedicated configuration for a specific board.
```

Comment:
Maybe we can mention here that these are called `Kconfig fragments`. We are already referring to these in the main documentation.